### PR TITLE
template: separate kubelet config into new MachineConfig

### PR DESCRIPTION
**Pull request by sjenning**
@abhinavdahiya @derekwaynecarr @gnufied @ashcrow 

Move the `/etc/kubernetes/kubelet.conf` and `/etc/systemd/system/kubelet.service` to a separate `MachineConfig`

### CODE REVIEW

**Comment by rphillips** [comment]
```
 got error on docker inspect fork/exec /bin/docker: no such file or directory
```
/retest

**Comment by rphillips** [comment]
@abhinavdahiya Do you know why this error might pop up?

```
CONFLICT (rename/delete): pkg/controller/template/test_data/templates/openstack/worker/files/-etc-hosts deleted in HEAD and renamed to pkg/controller/template/test_data/templates/worker/00-worker/openstack/files/-etc-hosts in 3ad4975b0b392073a6fa5dc63be9591a21114fc2. Version 3ad4975b0b392073a6fa5dc63be9591a21114fc2 of pkg/controller/template/test_data/templates/worker/00-worker/openstack/files/-etc-hosts left in tree.
CONFLICT (rename/delete): pkg/controller/template/test_data/templates/openstack/worker/files/-etc-docker-certs.d-docker-registry.default.svc:5000-ca.crt deleted in HEAD and renamed to pkg/controller/template/test_data/templates/worker/00-worker/openstack/files/-etc-docker-certs.d-docker-registry.default.svc:5000-ca.crt in 3ad4975b0b392073a6fa5dc63be9591a21114fc2. Version 3ad4975b0b392073a6fa5dc63be9591a21114fc2 of pkg/controller/template/test_data/templates/worker/00-worker/openstack/files/-etc-docker-certs.d-docker-registry.default.svc:5000-ca.crt left in tree.
CONFLICT (rename/delete): pkg/controller/template/test_data/templates/openstack/master/files/-etc-hosts deleted in HEAD and renamed to pkg/controller/template/test_data/templates/worker/00-worker/libvirt/files/-etc-hosts in 3ad4975b0b392073a6fa5dc63be9591a21114fc2. Version 3ad4975b0b392073a6fa5dc63be9591a21114fc2 of pkg/controller/template/test_data/templates/worker/00-worker/libvirt/files/-etc-hosts left in tree.
CONFLICT (rename/delete): pkg/controller/template/test_data/templates/openstack/master/files/-etc-docker-certs.d-docker-registry.default.svc:5000-ca.crt deleted in HEAD and renamed to pkg/controller/template/test_data/templates/worker/00-worker/libvirt/files/-etc-docker-certs.d-docker-registry.default.svc:5000-ca.crt in 3ad4975b0b392073a6fa5dc63be9591a21114fc2. Version 3ad4975b0b392073a6fa5dc63be9591a21114fc2 of pkg/controller/template/test_data/templates/worker/00-worker/libvirt/files/-etc-docker-certs.d-docker-registry.default.svc:5000-ca.crt left in tree.
```

**Comment by abhinavdahiya** [comment]
#225 removed some files.

**Comment by rphillips** [comment]
I'll rebase. Thanks!

**Comment by rphillips** [comment]
@sjenning If you reset your PR to https://github.com/rphillips/machine-config-operator/tree/add-kubelet-mcfg it'll fix the conflicts. (I don't have permission to push to your PR).

**Comment by sjenning** [comment]
rebased, see how it goes

File: pkg/controller/template/test_data/templates/master/01-master-kubelet/aws/units/kubelet.service [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        --rotate-certificates \
+        --kubeconfig=/var/lib/kubelet/kubeconfig \
+        --container-runtime=remote \
+        --container-runtime-endpoint=/var/run/crio/crio.sock \
+        --allow-privileged \
+        --node-labels=node-role.kubernetes.io/master \
+        --minimum-container-ttl-duration=6m0s \
+        --client-ca-file=/etc/kubernetes/ca.crt \
+        --cloud-provider=aws \
+        \
```
Comment (wking) [inline pkg/controller/template/test_data/templates/master/01-master-kubelet/aws/units/kubelet.service]: Why this empty line?  (spun off from [here][1])

[1]: https://github.com/openshift/installer/pull/1331#discussion_r261401041

**Pull request merged**